core/src/main/java/org/apache/iceberg/V1Metadata.java

```diff
@@ -356,7 +356,7 @@ public Object get(int pos) {
 
     @Override
     public void put(int i, Object v) {
-      throw new UnsupportedOperationException("Cannot read into IndexedDataFile");
+      throw new UnsupportedOperationException("Cannot write into IndexedDataFile");
```

Comment:
what is this change for?

Reply:
I think this `read` was a typo, unless I misunderstood this?

Reply:
This was correct before.

This class is only used to write v1 metadata files, not to read. So there is no need to implement the `put` method that is used when reading values from a file and putting them in the data file instance.

Reply:
Ah I see, it was "read into" to populate `IndexedDataFile`, makes sense. I'll revert this back.